src/common/match.c

```diff
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: GPL-2.0-only
+#define _POSIX_C_SOURCE 200809L
+#include <glib.h>
+#include <stdbool.h>
+#include "common/match.h"
+
+bool
+match(const gchar *pattern, const gchar *string)
+{
+	GString *p = g_string_new(pattern);
+	GString *s = g_string_new(string);
+	g_string_ascii_down(p);
+	g_string_ascii_down(s);
```

Comment:
Do we need a unicode down variant here? I am not sure if there are any restrictions to only use ASCII for class or app_id.

Reply:
Good shout. I think I'd like to add `title` at some point, so converting strings to form independent of case seems very sensible

It actually looks a lot simpler anyway :smile:

```
diff --git a/src/common/match.c b/src/common/match.c
index c4992e4..46de22c 100644
--- a/src/common/match.c
+++ b/src/common/match.c
@@ -7,13 +7,7 @@
 bool
 match(const gchar *pattern, const gchar *string)
 {
-	GString *p = g_string_new(pattern);
-	GString *s = g_string_new(string);
-	g_string_ascii_down(p);
-	g_string_ascii_down(s);
-	bool ret = (bool)g_pattern_match_simple(p->str, s->str);
-	g_string_free(p, true);
-	g_string_free(s, true);
-	return ret;
+	return (bool)g_pattern_match_simple(g_utf8_casefold(pattern, -1),
+		g_utf8_casefold(string, -1));
 }
 ```

Reply:
Change pushed.